### PR TITLE
Upgrade Go, Docker client and Alpine image

### PR DESCRIPTION
**Pull request by reneleonhardt**
* Upgrade Alpine 3.9.6 to 3.13.5
* Upgrade Docker 1.4.2 to 20.10.7
* Upgrade Go 1.13.3 to 1.16.5

Trivy showed several problems in the container image and dependencies, so I upgraded everything.
You can integrate it into GitHub Actions if you want:
https://aquasecurity.github.io/trivy/latest/examples/report/#sarif

### CODE REVIEW

**Comment by bsideup** [comment]
> Upgrade Docker 1.4.2 to 20.10.7

could you please make sure that the API Version stays the same?

**Comment by gesellix** [comment]
I think this one should be merged before #40. I'd rebase my pull request then.

**Comment by mdelapenya** [comment]
I'm working on a PR that adds unit tests using testcontainers-go for creating containers and networks to be removed. In that branch the version of the Docker dependency has been bumped, so it will get into conflicts with this one.

I think I'll send my PR today, so we will probably notify this PR with an update or even closing is as superseded

File: main.go [hunk 1]
```diff
@@ -74,8 +74,8 @@ func processRequests(deathNote map[string]bool, firstConnected chan<- bool, wg *
 			panic(err)
 		}
 		log.Printf("New client connected: %s\n", conn.RemoteAddr().String())
+		wg.Add(1)
```
Comment (mdelapenya) [inline main.go:77]: Could you send this potential fix including explanation in a separate PR? 🙏 

We are going to close this PR as the dependencies will be bumped in #46

**Comment by mdelapenya** [comment]
Given #50 is merged, and in there we are bumping:
- Upgrade Alpine to 3.16
- Upgrade Go to 1.18

I'm closing this PR as obsolete. In any case, thanks for contributing to Ryuk!

**Pull request closed without merging**